src/interface/params.hpp

```diff
@@ -50,11 +50,12 @@ class Params {
   }
 
   template <typename T>
-  const T &Get(const std::string &key) {
-    auto it = myParams_.find(key);
+  const T &Get(const std::string &key) const {
+    auto const it = myParams_.find(key);
     PARTHENON_REQUIRE_THROWS(it != myParams_.end(), "Key " + key + " doesn't exist");
-    PARTHENON_REQUIRE_THROWS(!(myTypes_[key].compare(std::string(typeid(T).name()))),
-                             "WRONG TYPE FOR KEY '" + key + "'");
+    PARTHENON_REQUIRE_THROWS(
+        !(myTypes_.find(key)->second.compare(std::string(typeid(T).name()))),
```

Comment:
should this be `.find()->second`? Or would `.at(key)` be better?

Reply:
Yes, it would be.   I'll make the change.

Reply:
Not only is it better, what I wrote is a bug because `find(key)` could return end which I then de referenced.  Doh!